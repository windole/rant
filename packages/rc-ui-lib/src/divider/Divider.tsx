import React, { useContext } from 'react';
import classnames from 'classnames';
import { DividerProps } from './PropsType';
import ConfigProviderContext from '../config-provider/ConfigProviderContext';

const Divider: React.FC<DividerProps> = ({
  children,
  className,
  hairline,
  dashed,
  contentPosition,
  ...props
}) => {
  const { prefixCls, createNamespace } = useContext(ConfigProviderContext);
  const [bem] = createNamespace('divider', prefixCls);

  return (
    <div
      role="separator"
      className={classnames(
        className,
        bem({
          dashed,
          hairline,
          [`content-${contentPosition}`]: !!children,
        }),
      )}
      {...props}
    >
      {children}
    </div>
  );
};

Divider.defaultProps = {
  hairline: true,
  contentPosition: 'center',
};

export default Divider;
