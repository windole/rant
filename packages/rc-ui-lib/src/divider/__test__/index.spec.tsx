import * as React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Divider from '..';

describe('Divider', () => {
  let wrapper;
  afterEach(() => {
    wrapper.unmount();
  });

  it('should render correctly', async () => {
    wrapper = mount(<Divider />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when have children', async () => {
    wrapper = mount(<Divider>text</Divider>);
    expect(wrapper.find('.rc-divider--content-center').text()).toEqual('text');
  });

  it('should render correctly when contentPosition is set for left', async () => {
    wrapper = mount(<Divider contentPosition="left">text</Divider>);
    expect(wrapper.find('.rc-divider--content-left').text()).toEqual('text');
  });

  it('should render correctly when contentPosition is set for right', async () => {
    wrapper = mount(<Divider contentPosition="right">text</Divider>);
    expect(wrapper.find('.rc-divider--content-right').text()).toEqual('text');
  });

  it('should render correctly when using dashed prop', async () => {
    wrapper = mount(<Divider dashed>text</Divider>);
    expect(wrapper.find('.rc-divider--dashed').text()).toEqual('text');
  });

  it('should render correctly when hairline is false ', async () => {
    wrapper = mount(<Divider>text</Divider>);
    expect(wrapper.find('.rc-divider--hairline').exists()).toBeTruthy();
    await wrapper.setProps({ hairline: false });
    expect(wrapper.find('.rc-divider--hairline').exists()).toBeFalsy();
  });
});
