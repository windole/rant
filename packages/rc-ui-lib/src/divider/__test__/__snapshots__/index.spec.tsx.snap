// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly 1`] = `
<Divider
  contentPosition="center"
  hairline={true}
>
  <div
    className="rc-divider rc-divider--hairline"
    role="separator"
  />
</Divider>
`;
